Puppi isolation miniAOD electrons & photons

ishvetso opened the pull request:
this PR refers to #15741. 

Changes:
-  to reco::Photon are disregarded as requested.
- Problem with associatedPackedPFCandidates() is solved in a different way. Isolation for the case of photons is computed based on particleFlow collection, and "reducedEgamma", "reducedPhotonPfCandMap" map is used for the footprint removal. For puppi weight is taken from the map produced by PuppiProducer that should be available during miniAOD. 
- particleBasedIsolation is configurable now in CITK.

matz-e committed:
"Seal" plugin declarations.

matz-e committed:
Fix include guard.

matz-e committed:
Move plugins into the right directory.

matz-e committed:
Spread SealModule.cc over the plugins to have them be more self-contained.

matz-e committed:
Fix more header guards.

matz-e committed:
Factor out more EDM plugin stuff and fix fallout.

matz-e committed:
Add lost plugin definitions.

ishvetso committed:
add puppi isolations to electron and photon objects

ishvetso committed:
add puppiNoLeptons as option + add fillDescriptions

ishvetso committed:
add puppi isolation to pat electron and photon producers

ishvetso committed:
update versions for PATPhotonProducer and PATElectronProducer

ishvetso committed:
remove puppi iso from PATPhotonSlimmer

ishvetso committed:
update configs for patElectrons and patPhotons for puppi iso sums

ishvetso committed:
fix inputs for puppi isolation for miniAOD

ishvetso committed:
remove puppi isolation sums from slimmedPhotons

ishvetso committed:
add CITK modules for PUPPI for electrons

bbockelm committed:
Report CPU usage information via HTCondor service.

This adds CPU usage information to the HTCondor service reporting;
it reports the entire CPU (user + system) usage for this process.
As the "job CPU" information reported by the framework excludes the
framework startup costs, the "total CPU" is slightly more than what
the Timing service prints on stdout.

bbockelm committed:
Report CPU model to HTCondor.

This commit has the framework forward the CPU model information to the
HTCondor monitoring.

bbockelm committed:
Include child processes in CPU timing.

With this commit, getCPU() additionally returns all usage information
from child processes.

jbsauvan committed:
Add HGCAL L1 trigger in SimL1Emulator sequence for phase2_hgcal era

jbsauvan committed:
Add HGCal trigger cells in event content

uhussain committed:
Fixed Dead Store-Dead assignment error in PATTriggerProducer.cc

ianna committed:
* Pixel nose description from Adrian

uhussain committed:
Fixed Dead Store-Dead assignment error in PATPackedCandidateProducer.cc

matz-e committed:
Add missing include which is now necessary.

matz-e committed:
Rename stray files.

matz-e committed:
Split new (un)packer and fix compilation.

matz-e committed:
Remove rebase remnant

uhussain committed:
Fixed Dead Store-Dead assignment error in JetCorrFactorsProducer.cc by removing unused variable

uhussain committed:
Fixed DeadStore-DeadAssignment errors in DCCSCBlock.cc

mtosi committed:
add orbit vs bx monitor

lgray committed:
address cosmetic code review

mmusich committed:
adding key for phase2 upgrade2023 MC simulation

jbsauvan committed:
Remove dummy backend clusters from event content

lgray committed:
put tracker customizations in the right eras

Commit:
Provide test beam simulation with beam spread

matz-e committed:
Fix hardcoded TDC mask for HF.

matz-e committed:
Provide better hardcoded values for LutMetadata.

Revert gain (which was changed in 6XY) back to 0.177 and provide better
values for rcalib etc corresponding to the 2017 era.

Commit:
Take care of Station status for GE21 chambers

Commit:
Merged refs/pull/15990/head from repository cms-sw

kpedro88 committed:
add HCAL to 2023 validation/DQM

lgray committed:
indents and copy/pasta

ishvetso committed:
Merged PUPPI_isolation_miniAOD_electrons_photons_v2 from repository ishvetso

kpedro88 committed:
make TP validation optional

kpedro88 committed:
remove noise validation for hcal phase2

Commit:
Add extra material for FNAL test beam setup

Commit:
Change debug flag

kpedro88 committed:
remove non-working hcal DQM modules for phase2

makortel committed:
Remove effectively unused methods

makortel committed:
Remove unneeded TrackingRegion parameter from SeedComparitor

This makes life easier for efficient regional CA seeding.

Commit:
Do not reset rechit flag inside the module

makortel committed:
Move theLayerCache.clear() to the end of hitTriplets/hitQuadruplets

CACell::getInnerHit() and getOuterHit() forward the calls to a
HitDoublets object, which forwards the calls to a RecHitsSortedInPhi
object owned by LayerHitMapCache. If the LayerHitMapCache is cleared,
the RecHitsSortedInPhi objects are deleted, so the clear() must be
done after all operations to HitDoublets.

VinInn committed:
add score plot

Commit:
Backup paths for inclusive high ET electrons

Commit:
Ad stage2 L1 trigger matching and possibility to sort matchings by L1 quality.

Commit:
fix number of stations

Commit:
change station 3 to 2

VinInn committed:
fix mistype

Commit:
fix GEM local reco step for GE21

Commit:
remove refs to station 3

mandrenguyen committed:
Add EPOS gen fragment for pPb relval

kpedro88 committed:
store QIE11 flag word for HE premixing

bsunanda committed:
Merge pull request #8 from dildick/from-CMSSW_8_1_X_2016-09-26-1500-Run2-gem19

Fixes to DIGI,L1,RECO and Validation steps with GE21

bbockelm committed:
Cleanly separate cpuinfo parsing from statistics collection.

Now, one function will parse the cpuinfo file - each different
statistic we want will utilize this information in a separate
method.

bbockelm committed:
Merge remote-tracking branch 'official-cmssw/CMSSW_8_1_X' into report_cpu_info

Conflicts:
	FWCore/Services/plugins/CPU.cc

Commit:
2-strip pad for GE21

VinInn committed:
reduce scale

jbsauvan committed:
Remove direct importing of Eras

VinInn committed:
protect against missing collection

makortel committed:
Fix CA customizes

SeedComparitor was not properly propagated.

Commit:
Detailed printout of the L1TMuonGlobalParams payload

Commit:
Updating the seeding CaloParams version

Commit:
Cosmetic changes to the standalone scripts

Martin-Grunewald committed:
81XHLTPrescaleProviderBugFix

uhussain committed:
Removed comment about flavor dependency on line238

gartung committed:
clang compilation error undefined reference to Phase2ITPixelCluster::MAXSPAN

Dr15Jones committed:
Only get data products with labels beginning with 'hlt'

When run in fastsim jobs, TriggerSummaryProducerAOD was requesting
data products for non HLT modules. Now it only accepts data products
from modules with labels beginning with 'hlt'. In addition, a new
parameter was added to the module to allow explicitly specifying
modules which should be avoided when choosing data products.

Dr15Jones committed:
Changed GeneratorSmearedProducer to be a global producer

kpedro88 committed:
fix premix config

kpedro88 committed:
keep presamples for qie8 in new packer

ianna committed:
* Merge name changes from dev materials
* Remove dev materials
* Remove obsolete file

Commit:
macros modified to save plots to root files

Commit:
Bug fix for IsolatedTrack analysis

Dr15Jones committed:
Use glob patterns when choosing module labels in TriggerSummaryProducerAOD

Use glob patterns to specify patterns to match and patterns to reject
for module labels when getting data products.

ishvetso committed:
add particleBasedIsolation configurable, fix PUPPI isolation source from reducedEgamma map

ishvetso committed:
fix inputs for miniAOD for PUPPI

cmsbuild commented:
A new Pull Request was created by  @ishvetso (Ivan Shvetsov) for CMSSW_8_1_X.

It involves the following packages:

DataFormats/PatCandidates
PhysicsTools/IsolationAlgos
PhysicsTools/PatAlgos
RecoEgamma/EgammaIsolationAlgos


@cmsbuild, @cvuosalo, @slava77, @monttj, @davidlange6 can you please review it and eventually sign? Thanks.
@rappoccio, @Sam-Harper, @jdolen, @imarches, @ahinzmann, @acaudron, @mmarionncern, @TaiSakuma, @gpetruc, @rafaellopesdesa, @JyothsnaKomaragiri, @nhanvtran, @gouskos, @gkasieczka, @schoef, @ferencek, @mverzett, @mariadalfonso, @lgray, @pvmulder, @cbernet this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

kpedro88 committed:
disable SiPM dark current noise during premixing step 1

cmsbuild commented:
-1

Tested at: af6a552e9d3e65d8914cde089f1ed150e39471f8

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16055/15478/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests  AddOn**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test runtestTqafExamples had ERRORS
---> test runtestTqafTopEventSelection had ERRORS
---> test runtestTqafTopEventProducers had ERRORS
---> test runtestTqafTopKinFitter had ERRORS
---> test runtestPhysicsToolsPatAlgos had ERRORS


* **AddOn**:

I found errors in the following addon tests:

cmsRun /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-09-29-2300/src/PhysicsTools/PatAlgos/test/IntegrationTest_cfg.py : FAILED - time: date Fri Sep 30 20:59:45 2016-date Fri Sep 30 20:59:08 2016 s - exit: 16640

ishvetso commented:
the error happens because the value map from the CITK is not found, and it is actually not run. It is not the case for my local test, and I think it is in the miniAOD_tools.py:
https://github.com/ishvetso/cmssw/blob/af6a552e9d3e65d8914cde089f1ed150e39471f8/PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py#L193

kpedro88 committed:
correctly set iEvent_ so txt files will not be produced

slava77 commented:
@ishvetso 
you can rerun the failing tests locally with ```scram b runtests```
The failures should be fixed before this PR can be merged

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16055/15478/summary.html

ishvetso commented:
@slava77  
local test gives the same problem. Modules are not included in the path. Should changes be made somewhere except PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py ?

slava77 commented:
On 9/30/16 2:02 PM, Ivan Shvetsov wrote:
> @slava77 <https://github.com/slava77>
> 
> local test gives the same problem. Modules are not included in the path.
> Should changes be made somewhere except
> PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py ?

The unit tests are for plain PAT, without miniAOD.
It's probably the easiest to just disable the puppi isolation in this case.


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16055#issuecomment-250851399>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcblVPHep3bVPJrB3Q3MVYroEyw6Jcks5qvXjOgaJpZM4KLTij>.
>

cmsbuild committed:
Merge pull request #16040 from gartung/DataFormats-Phase2ITPixelCluster-clang-compilation-fix

clang compilation error undefined reference to Phase2ITPixelCluster::MAXSPAN

cmsbuild committed:
Merge pull request #16045 from Dr15Jones/makeGeneratorSmearedProducerAGlobalModule

Changed GeneratorSmearedProducer to be a global producer

davidlange6 committed:
Merge pull request #16058 from kpedro88/StopEcalTxt

Stop producing ECAL txt files

cmsbuild committed:
Merge pull request #16006 from igv4321/HcalHitReconstructor-fix

Do not reset rechit flag inside the module

davidlange6 committed:
Merge pull request #15853 from uhussain/fix_staticAnaWarns

Fixing Dead store-Dead assignment errors in DCCSCBlock.cc, JetCorrFactorsProducer.cc, PATPackedCandidateProducer.cc & PATTriggerProducer.cc

ishvetso committed:
addPuppiIsolation added to configure

cmsbuild commented:
Pull request #16055 was updated. @cmsbuild, @cvuosalo, @slava77, @monttj, @davidlange6 can you please check and sign again.

davidlange6 committed:
delete what you create

ishvetso commented:
addPuppiIsolation option was added to patelectron and patphoton producers. My local test on pure miniAOD was successful, however there is a segmentation violation for unit test on PAT step. I'm not sure what the problem could be, but I was not sure about this line:
https://github.com/ishvetso/cmssw/blob/6ea103b0e4f34bb5d4ac0b5c91a9e2a2abff6114/PhysicsTools/PatAlgos/plugins/PATElectronProducer.cc#L922

cmsbuild commented:
The tests are being triggered in jenkins.

monttj commented:
In my local area with this PR at the latest 81X IB, I got segmentation violation in PATElectronProducer. 

---
Module: PATElectronProducer (crashed)

A fatal system signal has occurred: segmentation violation
Segmentation fault
---

Taejeong

ishvetso commented:
Hi Taejeong,

yes, that's what I also got. It might be due to this lines:

iDesc.ifValue(edm::ParameterDescription<bool>("addPuppiIsolation", 
false, true),
     true >> 
(edm::ParameterDescription<edm::InputTag>("puppiIsolationChargedHadrons", 
edm::InputTag("egmElectronPUPPIIsolation","h+-DR030-BarVeto000-EndVeto001"), 
true) and
edm::ParameterDescription<edm::InputTag>("puppiIsolationNeutralHadrons", 
edm::InputTag("egmElectronPUPPIIsolation","h0-DR030-BarVeto000-EndVeto000"), 
true) and
edm::ParameterDescription<edm::InputTag>("puppiIsolationPhotons", 
edm::InputTag("egmElectronPUPPIIsolation","gamma-DR030-BarVeto000-EndVeto008"), 
true)) or
     false >> edm::EmptyGroupDescription());

is this done correctly with the empty parameter description??

Ivan


01.10.16 17:42, Tae Jeong Kim пишет:
>
> In my local area with this PR at the latest 81X IB, I got segmentation 
> violation in PATElectronProducer.
>
> ------------------------------------------------------------------------
>
> Module: PATElectronProducer (crashed)
>
> A fatal system signal has occurred: segmentation violation
>
>
>     Segmentation fault
>
> Taejeong
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/16055#issuecomment-250918968>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFMRQsARcTL2_1IrlUJnWTXb8q3wyMxTks5qvn9ygaJpZM4KLTij>.
>

-- 
Best regards,
Ivan Shvetsov.

cmsbuild commented:
-1

Tested at: 6ea103b0e4f34bb5d4ac0b5c91a9e2a2abff6114

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16055/15484/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests  RelVals  AddOn**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test runtestTqafTopEventSelection had ERRORS
---> test runtestPhysicsToolsPatAlgos had ERRORS
---> test runtestTqafTopKinFitter had ERRORS
---> test runtestTqafExamples had ERRORS
---> test runtestTqafTopEventProducers had ERRORS


* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
135.4 step1
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step1_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

</pre>136.731 step3
<pre>runTheMatrix-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2023D1_GenSimFull+DigiFull_2023D1+RecoFullGlobal_2023D1+HARVESTFullGlobal_2023D1.log

</pre>20024.0 step3
<pre>runTheMatrix-results/20024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2023D1_GenSimFull+DigiFull_2023D1+RecoFullGlobal_2023D1+HARVESTFullGlobal_2023D1/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2023D1_GenSimFull+DigiFull_2023D1+RecoFullGlobal_2023D1+HARVESTFullGlobal_2023D1.log

</pre>10024.0 step3
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+HARVESTFull_2017.log

</pre>10424.0 step3
<pre>runTheMatrix-results/10424.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017NewFPix_GenSimFull+DigiFull_2017NewFPix+RecoFull_2017NewFPix+HARVESTFull_2017NewFPix/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017NewFPix_GenSimFull+DigiFull_2017NewFPix+RecoFull_2017NewFPix+HARVESTFull_2017NewFPix.log

</pre>10824.0 step3
<pre>runTheMatrix-results/10824.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017AllNew_GenSimFull+DigiFull_2017AllNew+RecoFull_2017AllNew+HARVESTFull_2017AllNew/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017HCALdev_GenSimFull+DigiFull_2017HCALdev+RecoFull_2017HCALdev+HARVESTFull_2017HCALdev.log

</pre>10624.0 step3
<pre>runTheMatrix-results/10624.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017HCALdev_GenSimFull+DigiFull_2017HCALdev+RecoFull_2017HCALdev+HARVESTFull_2017HCALdev/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017HCALdev_GenSimFull+DigiFull_2017HCALdev+RecoFull_2017HCALdev+HARVESTFull_2017HCALdev.log

</pre>50202.0 step3
<pre>runTheMatrix-results/50202.0_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50/step3_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50.log

</pre>25202.0 step3
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2023D3Timing_GenSimFull+DigiFull_2023D3Timing+RecoFullGlobal_2023D3Timing+HARVESTFullGlobal_2023D3Timing.log

</pre>22424.0 step3
<pre>runTheMatrix-results/22424.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2023D3Timing_GenSimFull+DigiFull_2023D3Timing+RecoFullGlobal_2023D3Timing+HARVESTFullGlobal_2023D3Timing/step3_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2023D3Timing_GenSimFull+DigiFull_2023D3Timing+RecoFullGlobal_2023D3Timing+HARVESTFullGlobal_2023D3Timing.log

</pre>

* **AddOn**:

I found errors in the following addon tests:

cmsRun /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-10-01-1100/src/PhysicsTools/PatAlgos/test/IntegrationTest_cfg.py : FAILED - time: date Sat Oct  1 20:23:31 2016-date Sat Oct  1 20:22:35 2016 s - exit: 35584
cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi --conditions auto:run2_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,HLT:@relval25ns,VALIDATION           --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot NominalCollision2015 --era Run2_25ns --magField 38T_PostLS1 : FAILED - time: date Sat Oct  1 20:23:47 2016-date Sat Oct  1 20:22:49 2016 s - exit: 23040
cmsDriver.py TTbar_13TeV_TuneCUETP8M1_cfi --conditions auto:run2_mc --fast  -n 100 --eventcontent AODSIM,DQM --relval 100000,1000 -s GEN,SIM,RECOBEFMIX,DIGI:pdigi_valid,L1,DIGI2RAW,L1Reco,RECO,EI,HLT:@relval2016,VALIDATION  --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --datatier GEN-SIM-DIGI-RECO,DQMIO --beamspot NominalCollision2015 --era Run2_2016 --magField 38T_PostLS1 : FAILED - time: date Sat Oct  1 20:35:04 2016-date Sat Oct  1 20:22:54 2016 s - exit: 23040

smuzaffar committed:
use v4 of HFShowerLibrary_npmt_noatt_eta4_16en

smuzaffar committed:
Merge pull request #16066 from smuzaffar/from-CMSSW_8_1_X_2016-10-01-1100

use v4 of HFShowerLibrary_npmt_noatt_eta4_16en

davidlange6 committed:
Merge pull request #16061 from davidlange6/fixl1tmem

Fix memory leak in L1Trigger/L1TMuonBarrel/src/L1MuBMEUX

cmsbuild committed:
Merge pull request #16031 from makortel/cleanSeedComparitor

Remove unused functionality from SeedComparitor

cmsbuild committed:
Merge pull request #16008 from makortel/fixCASegfault

Move LayerHitMapCache::clear() to very end in CA triplet and quadruplet generators

civanch committed:
fixed MC truth for SimHits

davidlange6 committed:
Merge pull request #16052 from friccita/MBValidationMacros

MB validation macros modified to save plots to root files

Commit:
Activate DB access for SiPM and TPs for HCAL

smuzaffar committed:
Add HGCSample as part of HGCDigi package

smuzaffar committed:
Merge pull request #16073 from smuzaffar/from-CMSSW_8_1_X_2016-10-02-2300

Add HGCSample as part of HGCDigi package

ianna committed:
* Remove modifications to 2016 scenario
* Remove Run2 materials

cmsbuild committed:
Merge pull request #16046 from Dr15Jones/onlyGetHLTDataForTriggerSummaryProducerAOD_2ndTry

Only get data products with labels beginning with 'hlt'

cmsbuild committed:
Merge pull request #15938 from lgray/followup_timing_codereview

Address residual code review for Timing PRs

cmsbuild committed:
Merge pull request #15975 from bsunanda/Run2-TB13

Run2-TB13 Provide test beam simulation with beam spread

cmsbuild committed:
Merge pull request #16000 from bsunanda/Run2-TB14

bsunanda:Run2-TB14  Add extra material for FNAL test beam setup

cmsbuild committed:
Merge pull request #16025 from dildick/from-CMSSW_8_1_0_pre10-GE21-2-strip-pad

2-strip pad for GE21

cmsbuild committed:
Merge pull request #16037 from Martin-Grunewald/81XHLTPrescaleProviderBugFix

81X HLTPrescaleProvider BugFix

cmsbuild committed:
Merge pull request #16069 from civanch/fixed_mc_truth

Fixed MC truth for SimHits

ishvetso committed:
fix switch for addPuppiIsolation

cmsbuild commented:
Pull request #16055 was updated. @cmsbuild, @cvuosalo, @slava77, @monttj, @davidlange6 can you please check and sign again.

Commit:
Components for validation using TB06 data

Commit:
Components for validation using TB06 data

ishvetso commented:
local unit tests successful, problem fixed.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

kpedro88 committed:
use shunt factor in QIE11 hardcode calibrations

lgray committed:
resurrecting old fasttiming from SLHC

cmsbuild commented:
+1
Tested at: adbc98a5f01365d6b9f088c6aa93c8865f82f6e0
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16055/15501/summary.html

lgray committed:
add geometry configs for D5 (D3 + timing layer)

kpedro88 committed:
newer TimeSlewSim for Hcal TB digis

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16055/15501/summary.html

lgray committed:
add D4/D5 to geometry dumpers

kpedro88 committed:
remove HcalHitCorrection entirely

matz-e committed:
Mention where these values come from.

lgray committed:
fix overlap, place barrel timing within Phase2OTBarrel partition

cmsbuild committed:
Merge pull request #16053 from bsunanda/Run2-alca64

bsunanda:Run2-alca64 Bug fix for IsolatedTrack analysis

cmsbuild committed:
Merge pull request #16038 from makortel/fixCACustomize

Fix to CA seeding customize functions

cmsbuild committed:
Merge pull request #15863 from mtosi/orbitVsBX81x

trigger monitoring : orbit vs bx

davidlange6 committed:
Merge pull request #15387 from matz-e/seal_81x

"Seal" L1T unpacking plugin declarations.

ishvetso committed:
add puppiNoLeptons for patElectrons

cmsbuild commented:
Pull request #16055 was updated. @cmsbuild, @cvuosalo, @slava77, @monttj, @davidlange6 can you please check and sign again.

ishvetso commented:
puppiNoLeptons is also added to patElectrons. That's the last change I was planning to introduce.

cmsbuild committed:
Merge pull request #16082 from kpedro88/Phase1-HE7

use shunt factor in QIE11 hardcode calibrations

cmsbuild committed:
Merge pull request #16072 from bsunanda/Run2-alca66

bsunanda:Run2-alca66 Activate DB access for SiPM and TPs for HCAL

rekovic committed:
Reduce memory usage by OMTF XMLConfigReader: (1) introduce desctructor, (2) release memory from parser back to system at the end of 'read' methods.

rekovic committed:
Fix valgrind errors of Invalid writes: reserve correct size of vector in L1TMuonBarrelParamsHelper.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

monttj commented:
@ishvetso 

I am not sure if this is really the solution. Because it is the same issue we had before if I understand the solution you provided correctly. 

It does not show the error simply because the value map you provided for the puppi isolation is not used anywhere. The default electron provided as input for PAT electron is not particle-flow electron so it does not go into the conditional statement where the puppi isolation is used for electron puppi isolation setup. 

I am wondering if you have checked that the final electron object has the puppi isolation with reasonable numbers? 

Regards,
Taejeong

ishvetso commented:
@monttj 
>It does not show the error simply because the value map you provided for the puppi isolation is not used anywhere. The default electron provided as input for PAT electron is not particle-flow electron so it does not go into the conditional statement where the puppi isolation is used for electron puppi isolation setup.

what lines you are referring to?

ianna committed:
Unify XML Materials for 2023

monttj commented:
I meant this line.
https://github.com/cms-sw/cmssw/pull/16055/files#diff-52916260b471455b5c3167c28fc7c9a4R317

This condition is always false.

Taejeong

ishvetso committed:
fix puppiIsolation with useParticleFlow option

cmsbuild commented:
Pull request #16055 was updated. @cmsbuild, @cvuosalo, @slava77, @monttj, @davidlange6 can you please check and sign again.

ishvetso commented:
Hi Taejeong,
thanks a lot for noticing this. It is fixed now.

cmsbuild committed:
Merge pull request #15952 from PFCal-dev/l1t-hgcal-in-std-sequences

Add HGCAL L1 trigger in SimL1Emulator and event content

cmsbuild committed:
Merge pull request #16080 from kpedro88/HcalRemoveHitCorrection

remove HcalHitCorrection entirely

cmsbuild committed:
Merge pull request #16077 from bsunanda/Run2-sim03

bsunanda:Run2-sim03 For validating G4 using 2006 test beam data

cmsbuild committed:
Merge pull request #16026 from VinInn/VtxDQM81

Add Score histos for Primary Vertices

cmsbuild commented:
Comparison not run due to Build errors (RelVals and Igprof tests were also skipped)

cmsbuild commented:
-1

Tested at: ad158ae76c679436cbaa5e4308efad0275cb3a17

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16055/15516/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found an error when building: 
 
 <pre>Leaving library rule at DQMServices/Core
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-10-04-1100/src/RecoMET/METPUSubtraction/plugins/JVFJetIdProducer.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-10-04-1100/src/RecoMET/METPUSubtraction/plugins/NoPileUpPFMEtDataProducer.cc 
>> Compiling edm plugin /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-10-04-1100/src/RecoEgamma/ElectronIdentification/plugins/cuts/GsfEleEInverseMinusPInverseCut.cc 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-10-04-1100/src/PhysicsTools/PatAlgos/plugins/PATElectronProducer.cc: In member function 'virtual void pat::PATElectronProducer::produce(edm::Event&, const edm::EventSetup&)':
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-10-04-1100/src/PhysicsTools/PatAlgos/plugins/PATElectronProducer.cc:361:18: error: 'class pat::Electron' has no member named 'setIsolationPUPPINoLeptons'
       anElectron.setIsolationPUPPINoLeptons((*PUPPINoLeptonsIsolation_charged_hadrons)[elePtr], (*PUPPINoLeptonsIsolation_neutral_hadrons)[elePtr], (*PUPPINoLeptonsIsolation_photons)[elePtr]);
                  ^
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-10-04-1100/src/PhysicsTools/PatAlgos/plugins/PATElectronProducer.cc:365:18: error: 'class pat::Electron' has no member named 'setIsolationPUPPINoLeptons'
       anElectron.setIsolationPUPPINoLeptons(-999., -999.,-999.);
                  ^
</pre>

* **Clang**:

I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 16 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>                   ^~~~~~~~~~~~~~~~
/cvmfs/cms-ib.cern.ch/2016-41/slc6_amd64_gcc530/external/gcc/5.3.0/lib/gcc/x86_64-pc-linux-gnu/5.3.0/../../../../include/c++/5.3.0/bits/atomic_base.h:157:26: note: expanded from macro 'ATOMIC_FLAG_INIT'
#define ATOMIC_FLAG_INIT { 0 }
                         ^~~~~
1 warning generated.
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-10-04-1100/src/PhysicsTools/PatAlgos/plugins/PATElectronProducer.cc:361:18: error: no member named 'setIsolationPUPPINoLeptons' in 'pat::Electron'
      anElectron.setIsolationPUPPINoLeptons((*PUPPINoLeptonsIsolation_charged_hadrons)[elePtr], (*PUPPINoLeptonsIsolation_neutral_hadrons)[elePtr], (*PUPPINoLeptonsIsolation_photons)[elePtr]);
      ~~~~~~~~~~ ^
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-10-04-1100/src/PhysicsTools/PatAlgos/plugins/PATElectronProducer.cc:365:18: error: no member named 'setIsolationPUPPINoLeptons' in 'pat::Electron'
      anElectron.setIsolationPUPPINoLeptons(-999., -999.,-999.);
      ~~~~~~~~~~ ^
</pre>

ishvetso committed:
add puppiNoLep to patElectron

cmsbuild commented:
Pull request #16055 was updated. @cmsbuild, @cvuosalo, @slava77, @monttj, @davidlange6 can you please check and sign again.

ishvetso commented:
puppiNoLep wasn't committed to pat::Electron, sorry.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Comparison not run due to Build errors (RelVals and Igprof tests were also skipped)

cmsbuild commented:
-1

Tested at: 5cf1c02b7c4338969748585f6dee58eb2f86660f

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16055/15524/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found an error when building: 
 
 <pre> static const boost::system::error_category& misc_category
                                             ^
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-10-04-1100/src/PhysicsTools/PatAlgos/plugins/JetCorrFactorsProducer.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-10-04-1100/src/PhysicsTools/PatAlgos/plugins/PATTauSlimmer.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-10-04-1100/src/PhysicsTools/PatAlgos/plugins/PATObjectFilter.cc 
error: class 'pat::Electron' has a different checksum for ClassVersion 34. Increment ClassVersion to 35 and assign it to checksum 905410242
Suggestion: You can run 'scram build updateclassversion' to generate src/DataFormats/PatCandidates/src/classes_def_objects.xml.generated with updated ClassVersion
gmake: *** [tmp/slc6_amd64_gcc530/src/DataFormats/PatCandidates/src/DataFormatsPatCandidates/libDataFormatsPatCandidates.so] Error 1
Leaving library rule at DataFormats/PatCandidates
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-10-04-1100/src/PhysicsTools/PatAlgos/plugins/CandidateSummaryTable.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_8_1_X_2016-10-04-1100/src/PhysicsTools/PatAlgos/plugins/PATMHTProducer.cc 
</pre>

ishvetso committed:
update version of patElectron

cmsbuild commented:
Pull request #16055 was updated. @cmsbuild, @cvuosalo, @slava77, @monttj, @davidlange6 can you please check and sign again.

ishvetso commented:
updated the version of pat::Electron. can you do another test? sorry

slava77 commented:
On 10/4/16 9:23 AM, Ivan Shvetsov wrote:
> updated the version of pat::Electron. can you do another test? sorry
> 

please confirm that you have this code working by making a fresh checkout.


> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16055#issuecomment-251438339>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcblKCSXZrJaqbmIHGUV3mnEA3oyWOks5qwn1lgaJpZM4KLTij>.
>

gartung committed:
Fix for clang compiler error missing symbol HGCalDetId::kHGCalCellMask

Commit:
Give RootEmbeddedFileSequence the same TTreeCache behavior as the primary file sequence, so input files for the MixingModule are cached

gartung committed:
Alignment/OfflineValidation : fix for clang compilation error missing symbols

cmsbuild committed:
Merge pull request #16091 from dan131riley/ttreecache-embeddedfileseq

Use TTreeCache for RootEmbeddedFileSequence

ishvetso commented:
confirmed.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

smuzaffar committed:
Update workflow 4.78 to use datasets from Run=207454

smuzaffar committed:
explicitly depend on gcc-atomic

This should explicitly linked libatomic from gcc. It should fix the link error for CLANG IBs
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc530/CMSSW_8_1_CLANG_X_2016-10-04-1100/FWCore/Services

cmsbuild commented:
+1
Tested at: 4076d88b5f8153d85caffa3eee2ed86b576ca472
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16055/15529/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16055/15529/summary.html

smuzaffar committed:
Merge pull request #16094 from cms-sw/smuzaffar-patch-6

Update workflow 4.78 to use datasets from Run=207454

cmsbuild committed:
Merge pull request #16096 from cms-sw/smuzaffar-patch-7

explicitly depend on gcc-atomic

cmsbuild committed:
Merge pull request #16012 from jruizvar/highEt81X

New version of backup paths for inclusive high ET electrons (81X)

cmsbuild committed:
Merge pull request #16016 from mandrenguyen/eposGenFragment_pPbRelVal_81X

Add EPOS gen fragment for pPb relval

ggovi committed:
Fixes for O2O steering

mmusich committed:
Add to phase-I queues updated version of SiPixelFedCablingMap for 48-channel feds, and update HCal QIEdata to version 2017_v2.0

gregor-mittag committed:
Temporary fix restricting offline-MillePede to single-IOV input.

monttj commented on PhysicsTools/PatAlgos/plugins/PATElectronProducer.cc:
```diff
@@ -323,6 +356,14 @@ void PATElectronProducer::produce(edm::Event & iEvent, const edm::EventSetup & i
+    else {
+      anElectron.setIsolationPUPPI(-999., -999.,-999.);
+      anElectron.setIsolationPUPPINoLeptons(-999., -999.,-999.);
+    }
```
@ishvetso 
Lastly, would it be possible for you to keep the indentation as other lines?

monttj commented on PhysicsTools/PatAlgos/plugins/PATElectronProducer.cc:
```diff
@@ -605,7 +647,15 @@ void PATElectronProducer::produce(edm::Event & iEvent, const edm::EventSetup & i
-
+      if (addPuppiIsolation_) {
+      anElectron.setIsolationPUPPI((*PUPPIIsolation_charged_hadrons)[elePtr], (*PUPPIIsolation_neutral_hadrons)[elePtr], (*PUPPIIsolation_photons)[elePtr]);
+      anElectron.setIsolationPUPPINoLeptons((*PUPPINoLeptonsIsolation_charged_hadrons)[elePtr], (*PUPPINoLeptonsIsolation_neutral_hadrons)[elePtr], (*PUPPINoLeptonsIsolation_photons)[elePtr]);
```
@ishvetso 
I would suggest to have some space inside the conditional statement for the indentation.

monttj commented on PhysicsTools/PatAlgos/plugins/PATElectronProducer.cc:
```diff
@@ -605,7 +647,15 @@ void PATElectronProducer::produce(edm::Event & iEvent, const edm::EventSetup & i
+      }
+      else {
+      anElectron.setIsolationPUPPI(-999., -999.,-999.);
+      anElectron.setIsolationPUPPINoLeptons(-999., -999.,-999.);
```
@ishvetso 
I would suggest to have some space inside the conditional statement for the indentation.

Commit:
Changes needed for IsoTrack validation (Data vs MC)

monttj commented on PhysicsTools/PatAlgos/plugins/PATElectronProducer.cc:
```diff
@@ -300,6 +332,7 @@ void PATElectronProducer::produce(edm::Event & iEvent, const edm::EventSetup & i
       bool MatchedToAmbiguousGsfTrack=false;
       for (edm::View<reco::GsfElectron>::const_iterator itElectron = electrons->begin(); itElectron != electrons->end(); ++itElectron) {
 	unsigned int idx = itElectron - electrons->begin();
+  auto elePtr = electrons -> ptrAt(idx);
```
@ishvetso
Would it be possible for you to keep the indentation as other lines?

monttj reviewed (commented):
I have a couple of comments for indentation.

ishvetso committed:
update indentation

cmsbuild commented:
Pull request #16055 was updated. @cmsbuild, @cvuosalo, @slava77, @monttj, @davidlange6 can you please check and sign again.

cmsbuild committed:
Merge pull request #16093 from gartung/Alignment-OfflineValidation-clang-error-fix

Alignment/OfflineValidation : Fix for clang compiler error undefined reference to PrimaryVertexValidation::nMaxtracks_

cmsbuild committed:
Merge pull request #16101 from cms-AlCaDB/81X_updateGlobalTagsForPre13

Add to phase-I queues updated version of SiPixelFedCablingMap for 48-channel feds, and update HCal QIEdata to version 2017_v2.0

rekovic committed:
Declare vector with correct size, and remove reserve().

rekovic committed:
Revert to reserve() for vector which is filled with calls to push_back().

gregor-mittag committed:
Added script to create single-IOV sqlite file for MPS.

monttj commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

ianna committed:
* Remove Tracker overlap with Beampipe
* Add Pixel nose
* Move 2015 dev Castor to prod

cmsbuild commented:
+1
Tested at: 37d7c4f6f47a69dec87f559c959f30fe2c5e4dfc
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16055/15565/summary.html

gregor-mittag committed:
Automatically create single-IOV input sqlite file in MPS.

Dr15Jones committed:
properly use eras with l1extraParticles

Moved the era modifications of l1extraParticles from
L1Trigger.Configuration.L1TReco_cff to here where they are meant
to be.
The default parameters for l1extraParticles now match the legacy trigger
values used in L1TReco_cff.

Dr15Jones committed:
Modernized use of eras and fixed problem with fastSim configuration

The 'isChosen' interface is deprecated for Modifiers so needed to
be replaced with with modern recommended usage for eras.

The module l1L1GtObjectMap never worked properly when run in fastSim
since it needed data from a module which was run in a later Path.
The result was the data from l1L1GtObjectMap was never put into the
event for fastSim. Now that the framework is going to run Paths
in parallel, the improper data dependency is producing an unrunnable
schedule. Therefore remove the module from the Sequence gives
functionally equivalent behavior as before this change.

gregor-mittag committed:
Replace iostream by edm logger in 'MillePedeAlignmentAlgorithm'.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16055/15565/summary.html

fabozzi committed:
new MC pPb wf based on EposLHC_4080_4080GeV

kpedro88 committed:
remove unnecessary default value

cmsbuild committed:
Merge pull request #16088 from cms-l1t-offline/pr81x-rebased-l1t-memoryFixes

Pr81x l1t memory fixes

davidlange6 committed:
Merge pull request #16117 from ianna/2017-overlaps-fix

2017 Overlaps Fix

davidlange6 committed:
Merge pull request #16011 from matz-e/hcal-tp-2017-hardcode

Update hardcoded LutMetadata for HCAL TP

davidlange6 committed:
Merge pull request #15992 from kpedro88/Phase2-WF20

add HCAL to 2023 validation/DQM

gregor-mittag committed:
Set PCL flag to 'True' in the harvesting step.

cmsbuild committed:
Merge pull request #16100 from ggovi/o2o-fixes-1-81X

Fixes for O2O steering

cmsbuild committed:
Merge pull request #16107 from bsunanda/Run2-alca67

bsunanda:Run2-alca67 Changes needed for IsoTrack validation (Data vs MC)

cmsbuild committed:
Merge pull request #16119 from fabozzi/from-CMSSW_8_1_X_HIrelval

new MC pPb wf based on EposLHC_4080_4080GeV

davidlange6 committed:
Merge pull request #16092 from gartung/L1Trigger-L1THGCal-clang-error-fix

L1Trigger/L1THGCal : Fix for clang compiler error undefined reference to HGCalDetId::kHGCalCellMask

davidlange6 committed:
Merge pull request #16028 from battibass/stage2_L1_matching_81X

Add stage2 L1 trigger matching + other features 81X

davidlange6 committed:
Merge pull request #16118 from Dr15Jones/properUseOfErasForL1Reco

Proper use of eras for L1Reco and a fastSim fix

davidlange6 committed:
Merge pull request #16034 from kkotov/postPR15842updates

L1T O2O: degubing of uGMT + new prototype for caloParams payload + scripts' development

monttj commented:
@ishvetso 
Comparing the default PF isolation, the puppi isolation seems correctly implemented.
![iso](https://cloud.githubusercontent.com/assets/4115236/19210998/cade7158-8d6c-11e6-9ea7-b83770410299.png)

However, it would be better to keep the same postfix so that we can avoid any mistake later due to the postfix changes.

So I would suggest lastly to change 
"puppiChargedHardonsIso()"--> "puppiChargedHadronIso()" (so no "s") 
same for neturalHaronIso and phtonIso.

I am sorry I just noticed this now. Could you provide this change once more?

Thank you for your effort to include this puppi isolation.

Regards,
Taejeong

ishvetso commented:
Hi Taejeong,

there is a significant difference in the first bin, I would expect charged isolation to be the same for standard PF isolation and puppi isolation. Do you agree?

Regards,
Ivan.

monttj commented:
You are right. I used default PF isolation which is currently 0.4 for electron. 
Once it is changed to 0.3, I have the same distribution as puppi for charged hadrons.
Small difference might come from the difference veto cone size.

![iso_charged](https://cloud.githubusercontent.com/assets/4115236/19213151/7a98fed4-8d9e-11e6-9e18-4f9a079c6fa6.png)
![iso_neutral](https://cloud.githubusercontent.com/assets/4115236/19213153/7c977c92-8d9e-11e6-9e58-5d0d06fbe329.png)
![iso_photon](https://cloud.githubusercontent.com/assets/4115236/19213154/7e769426-8d9e-11e6-8976-598c347596e1.png)

Last request from my side is to change the postfix: 

puppiChargedHardonsIso()--> puppiChargedHadronIso()
puppiNeutralHardonsIso()--> puppiNeutralHadronIso()
puppiPhotonsIso()-->puppiPhotonIso()

Hardons should be Hadron.

Taejeong

ishvetso committed:
fix postfix for puppi isolation

cmsbuild commented:
Pull request #16055 was updated. @cmsbuild, @cvuosalo, @slava77, @monttj, @davidlange6 can you please check and sign again.

ishvetso commented:
Hi Taejeong,

this should be fixed now. Thanks a lot for checking. I think veto cone is the same, the difference might come from this fromPV() function that is used to remove pileup.

Regards,
Ivan.

monttj commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: b30f52bc40b6c0a0fc890cdd106169f3fdbda909
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16055/15609/summary.html

cmsbuild committed:
Merge pull request #16134 from cms-AlCaDB/81X_revampPhase2GT

[81X] adding key for phase2 upgrade2023 MC simulation

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16055/15609/summary.html

davidlange6 committed:
Merge pull request #16057 from kpedro88/Phase1-HE6

Fixes for HCAL premixing

davidlange6 committed:
Merge pull request #15990 from bsunanda/Run2-gem19

bsunanda:Run2-gem19 Take care of Station status for GE21 chambers

Dr15Jones committed:
Fix summary information for Workers

The summary information for Workers was not properly thread-safe.
Switched to atomics but using loose memory ordering.

monttj commented:
+1
The indentation is not fixed yet completely.  
For example,
RecoEgamma/EgammaIsolationAlgos/python/egmPhotonIsolationPUPPI_cff.py

cmsbuild committed:
Merge pull request #15767 from bbockelm/report_cpu_info

Report CPU info to HTCondor monitoring

ishvetso committed:
fix indentation

cmsbuild commented:
Pull request #16055 was updated. @cmsbuild, @cvuosalo, @slava77, @monttj, @davidlange6 can you please check and sign again.

monttj commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 9212b2e8500671b9a46bb718b1b23d4500c455f5
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16055/15616/summary.html

cmsbuild committed:
Merge pull request #16158 from Dr15Jones/fixWorkerSummary

Fix summary information for Workers

cmsbuild committed:
Merge pull request #16135 from ghellwig/temporary-fix_multi-IOV-input_MillePede_81X

[81X] Temporary fix for multi-IOV input in MillePede

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16055/15616/summary.html

davidlange6 committed:
Merge pull request #16081 from lgray/phase2_timing_geometry

Phase2 Timing Layer Geometry

monttj commented:
+1

slava77 commented on DataFormats/PatCandidates/interface/Electron.h:
```diff
@@ -160,6 +160,29 @@ namespace pat {
+
+      float puppiNoLeptonsChargedHadronIso() const {return puppiNoLeptonsChargedHadronIso_; };
+      float puppiNoLeptonsNeutralHadronIso() const {return puppiNoLeptonsNeutralHadronIso_; };
+      float puppiNoLeptonsPhotonIso() const {return puppiNoLeptonsPhotonIso_; };
```
semicolon after the closing brace is superfluous.
Please remove.

slava77 commented on DataFormats/PatCandidates/interface/Photon.h:
```diff
@@ -169,6 +170,11 @@ namespace pat {
+
+      float puppiChargedHadronIso() const {return puppiChargedHadronIso_; };
+      float puppiNeutralHadronIso() const {return puppiNeutralHadronIso_; };
+      float puppiPhotonIso() const {return puppiPhotonIso_; };
```
drop the trailing semicolon

ishvetso replied:
fixed

slava77 commented on DataFormats/PatCandidates/interface/Photon.h:
```diff
@@ -322,6 +336,7 @@ namespace pat {
       reco::CandidatePtr sourceCandidatePtr( size_type i ) const;
 
       friend class PATPhotonSlimmer;
+      friend class AssociatedCandProducer;
```
is this needed?
I couldn't find a use case in this PR.

ishvetso replied:
fixed

slava77 commented on DataFormats/PatCandidates/interface/Photon.h:
```diff
@@ -169,6 +170,11 @@ namespace pat {
+      float puppiChargedHadronIso() const {return puppiChargedHadronIso_; };
+      float puppiNeutralHadronIso() const {return puppiNeutralHadronIso_; };
+      float puppiPhotonIso() const {return puppiPhotonIso_; };
+
```
why is there no NoLeptons version?
Was it forgotten or is there a subtle reason why it should not be here?

IIUC, the main feature of "NoLeptons" from perspective of ChargedHadronIso is that the leptons are not used in the input to the puppi algo (do not trigger de-weighting of a given particle as being consistent with pileup ... somewhat assuming that the leptons are not pileup).
So, from this, if it makes sense to do "NoLeptons" for electrons, it should be just about as meaningful for photons.

ishvetso replied:
The idea was to define puppi *without* candidates for which isolation is being calculated. Thus for electrons it is "NoLeptons", then for photons one has to define "NoPhotons" which is not defined clearly and we never studied that. Anyway, implementing this sort of thing is beyond this PR.

slava77 replied:
@nhanvtran @violatingcp 
please check this
I'd like to confirm that photon puppi iso will never have to be computed wrt puppiNoLeptons by design.
If we will need it in some analyses, then corresponding variables will need to be added to the pat::Photon

slava77 commented on DataFormats/PatCandidates/src/classes_def_objects.xml:
```diff
@@ -16,7 +16,10 @@
+  <class name="pat::Electron"  ClassVersion="36">
+   <version ClassVersion="36" checksum="3720919820"/>
+   <version ClassVersion="35" checksum="905410242"/>
+   <version ClassVersion="34" checksum="1857282274"/>
```
please remove all intermediate versions: there can not be more than one version value increment per PR
(drop 35 and 34; and then rename 36 to 35)
The same comment applies to the Photon.

ishvetso replied:
fixed

ishvetso replied:
@slava77  
actually in the newly created area I couldn't compile by just incrementing class version by +1.
it gets compiled with version 35 for pat::Electron and version 20 for pat::Photon
while in the release we have version 33 for pat::Electron and 18 for pat::Photon.
Should I do a commit back to these versions?

slava77 replied:
you should keep the checksum of what was 36 before and use it on v 34.
My instruction had a typo, it should have said ``` drop 35 and 34; and then rename 36 to 34 ``` (it should be followed in order).

ishvetso committed:
remove semicolon

cmsbuild commented:
Pull request #16055 was updated. @cmsbuild, @cvuosalo, @slava77, @monttj, @davidlange6 can you please check and sign again.

ishvetso committed:
update comments

cmsbuild commented:
Pull request #16055 was updated. @cmsbuild, @cvuosalo, @slava77, @monttj, @davidlange6 can you please check and sign again.

ishvetso commented on DataFormats/PatCandidates/interface/Electron.h:
```diff
@@ -344,6 +367,16 @@ namespace pat {
       float ecalPFClusIso_;
       float hcalPFClusIso_;
 
+      //PUPPI isolations
```
fixed

ishvetso committed:
remove unnecessary stuff

cmsbuild commented:
Pull request #16055 was updated. @cmsbuild, @cvuosalo, @slava77, @monttj, @davidlange6 can you please check and sign again.

ishvetso committed:
decrement version for patPhoton and patElectron

cmsbuild commented:
Pull request #16055 was updated. @cmsbuild, @cvuosalo, @slava77, @monttj, @davidlange6 can you please check and sign again.

ishvetso committed:
Merged PUPPI_isolation_miniAOD_electrons_photons_v2 from repository ishvetso

ishvetso committed:
update class version for patElectron and patPhoton

cmsbuild commented:
Pull request #16055 was updated. @ghellwig, @lveldere, @ianna, @cerminar, @fabozzi, @Martin-Grunewald, @rekovic, @vanbesien, @govoni, @perrotta, @civanch, @monttj, @silviodonato, @franzoni, @cmsbuild, @fwyzard, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @mdhildreth, @slava77, @ggovi, @mmusich, @mulhearn, @ssekmen, @perrozzi, @thuer, @dmitrijus, @alja, @srimanob, @hengne can you please check and sign again.

slava77 commented:
-1
needs a rebase or other steps to cleanup the history from irrelevant commits

ishvetso commented:
this is superseded by #16172
